tests/metrics/test_metric_definitions.py

```diff
@@ -175,8 +190,68 @@ def test_fails_to_load_metric():
     e = ImportError("mocked error")
     with mock.patch("evaluate.load", side_effect=e) as mock_load:
         with mock.patch("mlflow.metrics.metric_definitions._logger.warning") as mock_warning:
-            toxicity.eval_fn(eval_df, metrics={})
+            toxicity.eval_fn(eval_df["prediction"], None, {})
             mock_load.assert_called_once_with("toxicity", module_type="measurement")
             mock_warning.assert_called_once_with(
                 f"Failed to load 'toxicity' metric (error: {e!r}), skipping metric logging.",
             )
+
+
+def test_mae():
+    eval_df = pd.DataFrame({"prediction": [1.0, 2.0, 0.0], "target": [1.0, 2.0, 3.0]})
+    result = mae.eval_fn(eval_df["prediction"], eval_df["target"], {})
+    assert result.aggregate_results["mean_absolute_error"] == 1.0
+
+
+def test_mse():
+    eval_df = pd.DataFrame({"prediction": [1.0, 2.0, 0.0], "target": [1.0, 2.0, 3.0]})
+    result = mse.eval_fn(eval_df["prediction"], eval_df["target"], {})
+    assert result.aggregate_results["mean_squared_error"] == 3.0
+
+
+def test_rmse():
+    eval_df = pd.DataFrame({"prediction": [4.0, 5.0, 0.0], "target": [1.0, 2.0, 3.0]})
+    result = rmse.eval_fn(eval_df["prediction"], eval_df["target"], {})
+    assert result.aggregate_results["root_mean_squared_error"] == 3.0
+
+
+def test_r2_score():
+    eval_df = pd.DataFrame({"prediction": [1.0, 2.0, 3.0], "target": [3.0, 2.0, 1.0]})
+    result = r2_score.eval_fn(eval_df["prediction"], eval_df["target"], {})
+    assert result.aggregate_results["r2_score"] == -3.0
+
+
+def test_max_error():
+    eval_df = pd.DataFrame({"prediction": [1.0, 2.0, 3.0], "target": [3.0, 2.0, 1.0]})
+    result = max_error.eval_fn(eval_df["prediction"], eval_df["target"], {})
```

Comment:
Nit: Can we refactor these tests to not create eval_df and directly pass predictions and targets to the function?